packages/node-experimental/src/sdk/trace.ts

```diff
@@ -18,32 +19,26 @@ import type { NodeExperimentalClient } from '../types';
  * or you didn't set `tracesSampleRate`, this function will not generate spans
  * and the `span` returned from the callback will be undefined.
  */
-export function startSpan<T>(context: TransactionContext, callback: (span: Span | undefined) => T): T {
+export function startSpan<T>(spanContext: NodeExperimentalSpanContext, callback: (span: OtelSpan | undefined) => T): T {
   const tracer = getTracer();
   if (!tracer) {
     return callback(undefined);
   }
 
-  const name = context.name || context.description || context.op || '<unknown>';
-
-  return tracer.startActiveSpan(name, (span: OtelSpan): T => {
-    const otelSpanId = span.spanContext().spanId;
-
-    const sentrySpan = _INTERNAL_getSentrySpan(otelSpanId);
-
-    if (sentrySpan && isTransaction(sentrySpan) && context.metadata) {
-      sentrySpan.setMetadata(context.metadata);
-    }
+  const { name } = spanContext;
 
+  return tracer.startActiveSpan(name, (span): T => {
     function finishSpan(): void {
       span.end();
     }
 
+    _initSpan(span as OtelSpan, spanContext);
```

Comment:
q: I'm seeing quite a lot of type casting when dealing with otel spans. Is this because Otel types are somehow wrong/too broad?

Reply:
this is because `@opentelemetry/api` and everything related to that passes a basic `Span` type around, while we often need the spans that are actually generated by `@opentelemetry/sdk-trace-base` for certain things (because that has some more fields with things we need). But it's something we should look into when we stabilize this/split this up into better reusable parts, ideally we can avoid as much of this as possible 😬

Reply:
actually here specifically I'll update it and just pass the regular 'spans' around. that's safer and prob. more correct anyhow!